### PR TITLE
event log filter

### PR DESCRIPTION
**Pull request by filipesilva**
- feat: support event log migrations
- build: update to clojure 11
- rfct: move migrator into own ns
- fix: current version is 0 if none is present
- feat: migrate events without a uid
- fix: fix args on migrate call site
- rfct: add athens.self-hosted.fluree.utils
- fix: get-current-version should return 0 on err
- feat: migrate to efficient event log filtering

### CODE REVIEW

**Comment by filipesilva** [comment]
Tests hang forever both locally and on CI after adding the fluree tests. Opened an issue with repro at https://github.com/fluree/db/issues/163.

File: deps.edn [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {:paths ["src/clj" "src/cljs" "src/cljc" "src/js" "src/gen" "test"]
 
  :deps
- {org.clojure/clojure                   #:mvn{:version "1.10.3"}
+ {org.clojure/clojure                   #:mvn{:version "1.11.0"}
   org.clojure/clojurescript             #:mvn{:version "1.10.879"}
```
Comment (filipesilva) [inline deps.edn:5]: NB: clojurescript is not updated to 11 yet, that requires changing the version here and also a shadow-cljs version that supports it (iirc shadow-cljs overwrites it).

**Review by neotyk — APPROVED** [review]
LGTM

Tried shutting down JVM after tests, with little luck, Fluree is leaving a lot concurrent stuff behind.

File: src/clj/athens/self_hosted/event_log.clj [hunk 2]
```diff
@@ -39,52 +33,57 @@
...
 (defn serialize
   [id data]
+  (assert (uuid? id))
   {:_id :event
    :event/id (str id)
-   :event/data (pr-str data)})
+   :event/data (pr-str data)
+   ;; Compute the new order number as 1 higher than last.
+   ;; This will be 1 if there are no events yet.
+   ;; NOTE: is max-pred-val efficient for very large collections? I don't know.
```
[the remark this answers is not in the document]
Comment (filipesilva) [reply src/clj/athens/self_hosted/event_log.clj:42]: It is a smart function, used in a transaction: https://developers.flur.ee/docs/guides/advanced/smart-functions/10/

File: src/clj/athens/self_hosted/event_log.clj [hunk 3]
```diff
@@ -244,16 +240,18 @@
...
 
 (defn- recover-block-events
-  "Returns a seq of recovered events in conn for block=idx+1 in conn."
+  "Returns {:stop ... :next-page ... :items ...}, where items is a seq of recovered
+  events in conn for block=idx+1 in conn. For use with `iteration`."
   [conn idx]
-  (let [res @(fdb/block-query conn ledger {:block (inc idx) :pretty-print true})
+  (let [res @(fdb/block-query conn default-ledger {:block (inc idx) :pretty-print true})
         ex-msg (ex-message res)]
+    (println res)
```
[the remark this answers is not in the document]
Comment (filipesilva) [reply src/clj/athens/self_hosted/event_log.clj:248]: Wooops

[the remark this answers is not in the document]
Comment (filipesilva) [reply src/clj/athens/self_hosted/event_log.clj]: Fixed

**Pull request merged**